backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java

```diff
@@ -197,8 +198,46 @@ public ApiUser getCurrentApiUserInContainedTransaction() {
     return getCurrentApiUser();
   }
 
+  private ApiUser getPatientApiUser() {
+    TestOrder order = _contextHolder.getLinkedOrder();
+    if (order == null) {
+      return null;
+    }
+
+    PatientLink link = _patientLinkService.getFromTestOrder(order);
```

Comment:
We should just store the link itself in the patient context, not go around the block twice for it.